btp-document/BTP-Development-Resources.md

```diff
@@ -0,0 +1,49 @@
+
+- Interfaces of `BSH` on PRA (Solidity) [[link](BSH.md)]
+- Interface of `BSH` on ICON (Java) [[link](https://github.com/icon-project/btp/blob/master/doc/bsh.md)]
+
+## BTP Message Center (BMC)
+
+____
+
+BTP Message Center (BMC) is a smart contract that builds BTP Message before sending to the Relay (BMR) and handles relay messages from other BMRs. In addition, BMC also has features to manage authorization (e.g. Operators of a Contract and a list of allowance Relays), and information of connected networks (i.e. link, route) to make sure communication from the source delivering to the destination network can be operated smoothly and conveniently.
+
+- Interfaces of `BMC` on PRA (Solidity) [[link](BMC.md)]
+- Interface of `BMC` on ICON (Java) [[link](https://github.com/icon-project/btp/blob/master/doc/bmc.md)]
+
+## BTP Message Verifier (BMV)
+
+____
+
+BTP Message Verifier (BMV) is a smart contract that extracts a relay message, which is composed of a BTP message with proof of existence, to the BTP message before verifying. For easy verification, it may update trust information for the following events. Most of the implementations may track the hashes of block headers. If the blockchain system provides proof of absence of the BTP messages, then it's enough for the verifier to sustain the last state only. It updates the hash only if it sees proof of absence of further BTP messages in the block. Most blockchain systems don't provide proof of absence for their data. Therefore, it is mandatory to provide methods to verify historical hashes. Merkle Accumulator can be used for verifying old hashes. BMV sustains roots of Merkle Tree Accumulator, and relay will sustain all elements of Merkle Tree Accumulator. The relay may make the proof of any one of old hashes. So, even if byzantine relay updated the trust information with the proof of a new block, a normal relay can send BTP Messages in the past block with the proof.
+
+- Interfaces of `BMV` on PRA (Solidity) [[link](BMV.md)]
+- Interface of `BMV` on ICON (Java) [[link](https://github.com/icon-project/btp/blob/master/doc/bmv.md)]
+
+**Note that:** The provided interfaces are mandated to provide general or minimal requirements in building smart contracts so that distinct protocols could have compatible ways to interact and communicate with each other. Depend on the design of your project, a specific logic of implementations can be varied and additional functions/methods can be added. In the next section, we would like to give you a practical utilization of these interfaces and provide instructions in developing BTP Network connecting two blockchain ecosystems to a tee. Click on 'Next' if you are interested
```

Comment:
`Depend on the design of your project` -> `Depending on the design of your project`